python/GafferImageUI/ImageInspector.py

```diff
@@ -0,0 +1,460 @@
+			self._emitPathChanged()
+
+##########################################################################
+# _FormatPath
+##########################################################################
+
+class _FormatPath( _ImagePath ) :
+
+	def __init__( self, scriptNode, image, context, path, root = "/", filter = None ) :
+
+		_ImagePath.__init__( self, scriptNode, image, context, path, root, filter )
+
+	def propertyNames( self, canceller = None ) :
+
+		return Gaffer.Path.propertyNames( self ) + [ "format:value" ]
+
+	def property( self, name, canceller = None ) :
+
+		if name != "format:value" :
+			return Gaffer.Path.property( self, name, canceller )
+
+		if len( self ) != 1 :
+			return None
+
+		with Gaffer.Context( self._context, canceller ) :
+
+			if not self._image.channelNames() :
+				# Heuristic for detecting lack of input image.
+				return "---"
+
+			if self[0] == "Name" :
+				return GafferImage.Format.name( self._image.format() ) or "---"
+			elif self[0] == "Resolution" :
+				size = self._image.format().getDisplayWindow().size()
+				return f"{size.x} x {size.y}"
+			elif self[0] == "Pixel Aspect" :
+				return self._image.format().getPixelAspect()
+			elif self[0] == "Origin" :
+				origin = self._image.format().getDisplayWindow().min()
+				return f"{origin.x}, {origin.y}"
+			elif self[0] == "Data Window Origin" :
+				origin = self._image.dataWindow().min()
+				return f"{origin.x}, {origin.y}"
+			elif self[0] == "Data Window Size" :
+				size = self._image.dataWindow().size()
+				return f"{size.x} x {size.y}"
+
+	def _childNames( self, canceller ) :
+
+		if len( self ) :
+			return []
+
+		return [
+			"Name",
+			"Resolution",
+			"Pixel Aspect",
+			"Origin",
+			"Data Window Origin",
+			"Data Window Size",
+		]
+
+	def _imagePlugAffectsPath( self, plug ) :
+
+		return plug in [ self._image["format"], self._image["dataWindow"], self._image["channelNames"] ]
+
+##########################################################################
+# _MetadataPath
+##########################################################################
+
+class _MetadataPath( _ImagePath ) :
+
+	def __init__( self, scriptNode, image, context, path, root = "/", filter = None ) :
+
+		_ImagePath.__init__( self, scriptNode, image, context, path, root, filter )
+
+	def propertyNames( self, canceller = None ) :
+
+		return Gaffer.Path.propertyNames( self ) + [ "metadata:value" ]
+
+	def property( self, name, canceller = None ) :
+
+		if name == "metadata:value" :
+			return self.__metadata( canceller ).get(
+				str( self )[1:]
+			)
+
+		return Gaffer.Path.property( self, name, canceller )
+
+	def _childNames( self, canceller ) :
+
+		metadata = self.__metadata( canceller )
+		selfNames = self[:]
+
+		result = []
+		visited = set()
+		for key in sorted( metadata.keys() ) :
+			names = key.split( "/" )
+			if len( names ) <= len( selfNames ) or names[:len(selfNames)] != selfNames :
```

Comment:
Do you think it would be worth adding a comment to explain this without requiring prior
knowledge about `Gaffer::Path`?

Reply:
I think it's reasonable to expect that anyone maintaining this code should know (or will look up) that paths use list syntax to access the names. We can't document fundamentals at every point of usage.